src/test/java/com/wikia/webdriver/testcases/mercurytests/CategoryPageTest.java

```diff
@@ -1,56 +1,123 @@
   }
 
   @Test(groups = "mercury_category_navigateToCategoryPageWithArticleAndWithoutMembersFromUrl")
-  @RelatedIssue(issueID = "SEO-618", comment = "commit was reverted;tests- to be updated")
   public void mercury_category_navigateToCategoryPageWithArticleAndWithoutMembersFromUrl() {
-    new CategoryPage().navigateToPageWithArticleAndWithoutMembersFromUrl();
+    final String categoryName = String.format("Category:%s", TestContext.getCurrentMethodName());
+
+    new ArticleContent().push("some irrelevant content of category article", categoryName);
+    CategoryPage category = new CategoryPage().open(categoryName);
+
+    Assert.assertEquals(404, category.getURLStatus(category.getCurrentUrl()));
   }
 
   @Test(groups = "mercury_category_navigateToCategoryPageWithArticleAndWithoutMembersFromLink")
   public void mercury_category_navigateToCategoryPageFromLinkInArticle() {
-    new ArticlePage()
-        .openPageWithLinkToCategoryPage()
+    final String categoryName = String.format("Category:%s", TestContext.getCurrentMethodName());
+
+    new ArticleContent().push(String.format("[[%s]]", categoryName));
+
+    CategoryPage categoryPage = new ArticlePage()
+        .open()
         .openCategoryPageFromLink();
-  }
 
-  @Test(groups = "mercury_category_navigateThroughSection")
-  public void mercury_category_navigateThroughSection() {
-    new CategoryPage()
-        .navigateToPageWithArticleAndWithMembersFromUrl()
-        .loadMoreMembersForSection("C")
-        .loadPreviousMembersForSection("C");
+    Assert.assertTrue(categoryPage.hasCategoryMembers());
   }
 
   @Test(groups = "mercury_category_navigateToCategoryMemberPage")
   public void mercury_category_navigateToCategoryMemberPage() {
-    new CategoryPage()
-        .navigateToPageWithArticleAndWithMembersFromUrl()
-        .navigateToCategoryMemberPage("Category test 001");
+    final String categoryName = String.format("Category:%s", TestContext.getCurrentMethodName());
+    final String articleContent = "Test article content";
+
+    new ArticleContent().push(String.format("%s [[%s]]", articleContent, categoryName));
+
+    ArticlePage article = new CategoryPage()
+        .open(categoryName)
+        .navigateToCategoryMemberPage();
+
+    Assert.assertEquals(articleContent, article.getArticleContent());
+  }
+
+  @Test(groups = "mercury_category_testPageination")
+  public void mercury_category_testPagination() {
+    // This test assumes that there are more than 200 articles created and with category like one below.
+    // Staging env should have fixture of this articles to create them everytime it is rebuilded
+
+    final String categoryName = String.format("Category:%s", TestContext.getCurrentMethodName());
+    ArticleContent articleContent = new ArticleContent();
+
+    articleContent.push("some irrelevant content of category article", categoryName);
+
+    // execute this loop only once on staging and then create fixture
+//    for (int i = 0; i < 300; ++i) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3ACommentedOutCodeLine)